### PR TITLE
[Datasets] Warn user if missing columns and support `str` `exclude`

### PR DESCRIPTION
**Pull request by bveeramani**
Signed-off-by: Balaji <balaji@anyscale.com>

## Why are these changes needed?

Other preprocessors support specifying columns with a single string. For example, you'd expect that you'd be able to do
```
preprocessor = Concatenator(exclude="label")
```

In reality, `Concatenator` tries to exclude columns named `"l"`, `"a"`, `"b"`, and so on. This leads to unexpected behavior.

This PR:
1. Adds support for string-valued `exclude` arguments
2. Adds a warning when the user specifies non-existant columns

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/preprocessors/concatenator.py [hunk 1]
```diff
@@ -108,34 +113,34 @@ def __init__(
...
-                raise ValueError(
-                    f"Missing columns specified in 'include': {missing_columns}"
-                )
+    def _validate(self, df: pd.DataFrame) -> None:
+        for parameter in "include", "exclude":
+            columns = getattr(self, parameter)
+            if columns is None:
+                continue
+
+            missing_columns = set(columns) - (set(df) & set(columns))
```
Comment (clarkzinzow) [inline python/ray/data/preprocessors/concatenator.py:135]: Nit: Taking the intersection for the set difference is redundant, taking the direct difference will work as expected.
```suggestion
            missing_columns = set(columns) - set(df)
```

**Review by clarkzinzow — APPROVED** [review]
LGTM!

File: python/ray/data/preprocessors/concatenator.py [hunk 2]
```diff
@@ -108,34 +113,34 @@ def __init__(
...
-                raise ValueError(
-                    f"Missing columns specified in 'include': {missing_columns}"
-                )
+    def _validate(self, df: pd.DataFrame) -> None:
+        for parameter in "include", "exclude":
+            columns = getattr(self, parameter)
+            if columns is None:
+                continue
+
+            missing_columns = set(columns) - set(df)
```
[the remark this answers is not in the document]
Comment (clarkzinzow) [reply python/ray/data/preprocessors/concatenator.py:135]: That's a good point!

**Comment by richardliaw** [comment]
Is this PR ready to merge?

**Comment by bveeramani** [comment]
> Is this PR ready to merge?

Should be after CI runs.

File: python/ray/data/preprocessors/concatenator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bveeramani) [reply python/ray/data/preprocessors/concatenator.py:139]: Addressed

**Pull request merged**